dashboard/app/models/script.rb

```diff
@@ -333,8 +333,9 @@ def self.lesson_extras_script_ids
     @@lesson_extras_script_ids ||= all_scripts.select(&:lesson_extras_available?).pluck(:id)
   end
 
-  def self.maker_units
-    @@maker_units ||= visible_units.select(&:is_maker_unit?)
+  def self.maker_units(user)
+    return_units = @@maker_units ||= visible_units.select(&:is_maker_unit?)
```

Comment:
(minor, up to you) Now that I see this, I'm wondering whether this would be clearer as two separate functions `maker_units` and `maker_units_for_user` or `visible_maker_units` and `maker_units` or something like that.

Reply:
I see what you mean here. I think if this were used in more places, the clarity would help/make sense. That said, the function is only called from one place so I think it's fine to keep them combined.